kanidmd/src/lib/macros.rs

```diff
@@ -240,41 +240,77 @@ macro_rules! run_create_test {
         use crate::prelude::*;
         use crate::schema::Schema;
         use crate::utils::duration_from_epoch_now;
-
-        let mut au = AuditScope::new("run_create_test", uuid::Uuid::new_v4(), None);
-        lperf_segment!(&mut au, "plugins::macros::run_create_test", || {
-            let qs = setup_test!(&mut au, $preload_entries);
-
-            let ce = match $internal {
-                None => CreateEvent::new_internal($create_entries.clone()),
-                Some(e_str) => unsafe {
-                    CreateEvent::new_impersonate_entry_ser(e_str, $create_entries.clone())
-                },
-            };
-
-            {
-                let qs_write = qs.write(duration_from_epoch_now());
-                let r = qs_write.create(&mut au, &ce);
-                debug!("test result: {:?}", r);
-                assert!(r == $expect);
-                $check(&mut au, &qs_write);
-                match r {
-                    Ok(_) => {
-                        qs_write.commit(&mut au).expect("commit failure!");
+        use crate::logbuffer::LogBuffer;
+
+        use tracing::{self, Level};
+        use tracing_subscriber::FmtSubscriber;
+        use tracing_subscriber::fmt::format::{FmtSpan, debug_fn};
+        use tracing_subscriber::field::MakeExt;
+
+        // https://docs.rs/tracing-subscriber/0.2.18/tracing_subscriber/fmt/struct.SubscriberBuilder.html
+        let subscriber = FmtSubscriber::builder()
+            // SETTINGS
+            .with_max_level(Level::TRACE)
+            .with_ansi(false) // no colors (good for .log files)
+
+            // JSON
+            // .json() // machine-readable
+            // .with_span_list(false) // displays all open spans
+
+            .with_writer(LogBuffer::default())
+            .with_span_events(FmtSpan::NEW | FmtSpan::CLOSE) // logs when `span`s are initialized or terminated
+
+            .fmt_fields({
+                debug_fn(|writer, key, value| {
+                    match key.to_string().as_str() {
+                        "message" => write!(writer, "{:?}", value),
+                        _ => write!(writer, "{}={:?}", key, value),
                     }
-                    Err(e) => {
-                        ladmin_error!(&mut au, "Rolling back => {:?}", e);
+                }).delimited(" | ")
+            })
+
+            .finish();
+
+        tracing::subscriber::with_default(subscriber, || {
```

Comment:
Instead of using FmtSubscriber, why not create https://docs.rs/tracing/0.1.26/tracing/trait.Subscriber.html? This would let us capture all the data we want using tracing / spans, but then format and display it how we like instead. 

IE I think the solution here is what we want is to use tracing's machineries/facades, but then capture those events and process them in a way that is kanidm specific, and managing how we output these to a file/stdout/stderr.

In a way it feels like trying to capture the "output" buffers and trying to sync/manage that is working at the wrong level for the problem, when what we really want is to capture tracings *events* and then buffering these in a native format for post-processing.